src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,143 @@
   const notify = useNotify();
-  const { canViewResources } = useServerEntitlements();
+  const isClustered = useIsClustered();
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const currentServer = (() => {
+    const createTarget =
+      (values as CreateInstanceFormValues)?.target ?? undefined;
+    if (createTarget && !createTarget.startsWith(CLUSTER_GROUP_PREFIX)) {
+      return createTarget;
+    }
+    const location = (values as EditInstanceFormValues)?.location ?? undefined;
+    return location === "none" || !location ? undefined : location;
+  })();
 
   const {
     data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
+    error: error,
+    isLoading: isLoading,
+  } = useResources(clusterMembers, currentServer);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
+    return null;
   }
 
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+  const getAvailableMemory = (resources: LxdResources[]): MemoryLimit => {
+    const sortedResourcesMemories = resources
+      .map((r) => r.memory.total)
+      .sort((a, b) => a - b);
+    const minResourceMemory = sortedResourcesMemories[0];
+    const maxResourceMemory =
+      sortedResourcesMemories[sortedResourcesMemories.length - 1];
+    if (!projectResourceMemory) {
+      return {
+        range: [minResourceMemory, maxResourceMemory],
+        source: isClustered && currentServer ? "cluster-member" : "",
+      };
     }
-    if (!resources?.memory.total) {
-      return limitToBytes(project.config["limits.memory"]);
+    const projectLimit = Number(projectResourceMemory);
+
+    if (sortedResourcesMemories.length === 0) {
+      return {
+        range: [projectLimit, projectLimit],
+        source: "project",
+      };
     }
-    return Math.min(
-      resources.memory.total,
-      Number(limitToBytes(project.config["limits.memory"])),
-    );
+    return {
+      range: [
+        Math.min(minResourceMemory, projectLimit),
+        Math.min(maxResourceMemory, projectLimit),
+      ],
+      source: "",
+    };
   };
 
-  const maxMemory = getAvailableMemory();
+  const {
+    range: [minMemory, maxMemory],
+    source,
+  } = getAvailableMemory(resources ?? []);
+
+  let resourceLink = null;
+  if (source === "project") {
+    resourceLink = (
+      <ResourceLink
+        type="project"
+        value={project?.name || "default"}
+        to={`/ui/projects/${encodeURIComponent(project?.name || "default")}`}
+      />
+    );
+  } else if (source === "cluster-member" && currentServer) {
```

Comment:
```suggestion
  } else if (source === "cluster-member") {
```
current server is defined in this case.